src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -472,38 +500,16 @@ export function validateComponentFormData(
   return null;
 }
 
-export function getSchemaPart(dataModelPath: string[], subSchema: any, mainSchema: any) {
-  const dataModelRoot = dataModelPath[0];
-  if (subSchema.properties && subSchema.properties[dataModelRoot] && dataModelPath && dataModelPath.length !== 0) {
-    const localRootElement = subSchema.properties[dataModelRoot];
-    if (localRootElement.$ref) {
-      const childSchemaPtr = JsonPointer.compile(localRootElement.$ref.substr(1));
-      return getSchemaPart(dataModelPath.slice(1), childSchemaPtr.get(mainSchema), mainSchema);
-    }
-    if (localRootElement.items && localRootElement.items.$ref) {
-      const childSchemaPtr = JsonPointer.compile(localRootElement.items.$ref.substr(1));
-      return getSchemaPart(dataModelPath.slice(1), childSchemaPtr.get(mainSchema), mainSchema);
-    }
-    return localRootElement;
-  }
-
-  if (subSchema.allOf) {
-    let tmpSchema: any = {};
-    subSchema.allOf.forEach((element) => {
-      tmpSchema = {
-        ...tmpSchema,
-        ...element,
-      };
-    });
-    return getSchemaPart(dataModelPath, tmpSchema, mainSchema);
-  }
-
-  if (subSchema.$ref) {
-    const ptr = JsonPointer.compile(subSchema.$ref.substr(1));
-    return getSchemaPart(dataModelPath.slice(1), ptr.get(mainSchema), mainSchema);
+export function getSchemaPart(schemaPath: string, jsonSchema: object): any {
+  // comes at format "#/properties/model/properties/person/properties/name/maxLength"
```

Comment:
Maybe the comment here could be over the exported string as apart of a ts-doc so users know that the schemaPath is supposed to start with a `#`?

Reply:
Also an assertion for that would make sense

Reply:
Great suggestion. Fixed.